kratos/includes/constitutive_law.h

```diff
@@ -394,11 +396,11 @@ class KRATOS_API(KRATOS_CORE) ConstitutiveLaw : public Flags
       void SetShapeFunctionsValues         (const Vector& rShapeFunctionsValues)      {mpShapeFunctionsValues=&rShapeFunctionsValues;};
       void SetShapeFunctionsDerivatives    (const Matrix& rShapeFunctionsDerivatives) {mpShapeFunctionsDerivatives=&rShapeFunctionsDerivatives;};
 
-      void SetDeformationGradientF         (const Matrix& rDeformationGradientF)      {mpDeformationGradientF=&rDeformationGradientF;};
+      void SetDeformationGradientF         (const DeformationGradientMatrixType& rDeformationGradientF)      {mpDeformationGradientF=&rDeformationGradientF;};
```

Comment:
this is another class, although defined within the ConstitutiveLaw class.

i wonder if you should not use "ConstitutiveLaw::DeformationGradientMatrixType"

Reply:
(same in the next few entries)